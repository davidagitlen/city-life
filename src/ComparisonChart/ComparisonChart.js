import React, { Component } from 'react';
import { connect } from 'react-redux';
import './ComparisonChart.scss';

class ComparisonChart extends Component{
  chartRef = React.createRef();
  // constructor(){
  //   super();

  // }

  componentDidMount() {
    const currentChartRef = this.chartRef.current.getContext('2d');

    // new Chart(currentChartRef, {
    //   type: 'bar',
    //   data: {
    //     labels: currentSleepUser.sleepPropertyOneWeek('2019/09/16', '2019/09/22', 'hoursSlept').map(day => day[0].slice(5)),
    //     datasets: [
    //       {
    //         label: 'Last Week\'s Hours of Sleep',
    //         data: currentSleepUser.sleepPropertyOneWeek('2019/09/16', '2019/09/22', 'hoursSlept').map(day => day[1]),
    //         backgroundColor: [
    //           'rgba(148, 0, 211, .25)',
    //           'rgba(148, 0, 211, .25)',
    //           'rgba(148, 0, 211, .25)',
    //           'rgba(148, 0, 211, .25)',
    //           'rgba(148, 0, 211, .25)',
    //           'rgba(148, 0, 211, .25)',
    //           'rgba(148, 0, 211, .25)',
    //         ],
    //         borderColor: [
    //           'rgba(128, 0, 128, 1)',
    //           'rgba(128, 0, 128, 1)',
    //           'rgba(128, 0, 128, 1)',
    //           'rgba(128, 0, 128, 1)',
    //           'rgba(128, 0, 128, 1)',
    //           'rgba(128, 0, 128, 1)',
    //           'rgba(128, 0, 128, 1)',
    //         ],
    //         borderWidth: 1,
    //         yAxisId: 'Sleep Hours'
    //       },
    //       {
    //         type: 'line',
    //         label: '  Last Week\'s Sleep Quality',
    //         data: currentSleepUser.sleepPropertyOneWeek('2019/09/16', '2019/09/22', 'sleepQuality').map(day => day[1]),
    //         backgroundColor: 'rgba(0, 0, 0, 0)',
    //         borderColor: 'rgba(128, 0, 128, 1)',
    //         yAxisID: 'Sleep Quality',
    //         borderWidth: 1.5
    //       },
    //     ]
    //   },
    //   options: {
    //     responsive: false,
    //     scales: {
    //       yAxes: [{
    //         id: 'Sleep Hours',
    //         ticks: {
    //           beginAtZero: true,
    //         },
    //         scaleLabel: {
    //           display: true,
    //           labelString: 'Hours'
    //         }
    //       },
    //       {
    //         id: 'Sleep Quality',
    //         position: 'right',
    //         ticks: {
    //           beginAtZero: true,
    //         },
    //         scaleLabel: {
    //           display: true,
    //           labelString: 'Sleep Quality'
    //         }
    //       },
    //       ]
    //     }
    //   }
    // });

  }

  render() {
    console.log(this.props)
    return(
      <>
      </>
    )
  }

}



export default ComparisonChart;