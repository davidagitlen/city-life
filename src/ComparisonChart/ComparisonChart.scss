.ComparisonChart {
  height: 400px;
  width: 400px;
  font-size: 16px;
  margin: 15px 0px 0px 15px;
  background-color: rgba(46,46,201, .05);
}