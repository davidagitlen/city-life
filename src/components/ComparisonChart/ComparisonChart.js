import React, { Component } from 'react';
import Chart from 'chart.js';
import './ComparisonChart.scss';

class ComparisonChart extends Component{
  chartRef = React.createRef();

  componentDidMount() {
    this.createChart();
  }

  componentDidUpdate() {
    this.createChart();
  }

  createChart = () => {
    const { data, cityOneName, cityTwoName } = this.props;
    const currentChartRef = this.chartRef.current.getContext('2d');

    new Chart(currentChartRef, {
      type: 'line',
      data: {
        labels: data.cityOneData.map(datum => datum.name),
        datasets: [
          {
            label: cityOneName,
            data: data.cityOneData.map(datum => datum.score_out_of_10.toFixed(2)),
            fill: false,
            backgroundColor: data.cityOneData.map(datum => '#2E2EC9'),
            borderColor: data.cityOneData.map(datum => '#2E2EC9'),
            borderWidth: 1
          },
          {
            type: 'line',
            label: cityTwoName,
            data: data.cityTwoData.map(datum => datum.score_out_of_10.toFixed(2)),
            fill: false,
            backgroundColor: data.cityTwoData.map(datum => '#6DECAF'),
            borderColor: data.cityTwoData.map(datum => '#6DECAF'),
            borderWidth: 1.5
          },
        ]
      },
      options: {
        responsive: true,
        maintainAspectRatio: false,
        scales: {
          yAxes: [{
            ticks: {
              beginAtZero: true,
              suggestedMax: 10,
            },
            gridLines: {
              display: true,
              drawBorder: false
            },
            scaleLabel: {
              display: true,
              labelString: 'Score'
            }
          }
          ]
        }
      }
    });
  }

  render() {
    console.log('in comparisonChart', this.props)
    return(
        <div className='ComparisonChart'>
          <canvas
            id="myChart"
            ref={this.chartRef}
          />
        </div>
    )
  }
}

export default ComparisonChart;