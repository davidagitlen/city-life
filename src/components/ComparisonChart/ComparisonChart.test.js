import React from 'react';
import ComparisonChart from './ComparisonChart';
import { shallow } from 'enzyme';

describe('ComparisonChart', () => {

  let wrapper, mockData;
  // global.createRef() = jest.fn().mockImplementation(() => ({
  //   current : {
  //     getContext : jest.fn();
  //   }
  // }));

  beforeEach(() => {
    mockData = {
      cityOneData: [{ name: 'Housing', score_out_of_10: 10 }, { name: 'Startups', score_out_of_10: 10 }],
      cityTwoData: [{ name: 'Housing', score_out_of_10: 5 }, { name: 'Startups', score_out_of_10: 5 }]
    };

    wrapper = shallow(
      <ComparisonChart 
        data={mockData}
        cityOneName='Boulder'
        cityTwoname='Topeka'
      />
    );
  });

  it.skip('should match the snapshot with data passed in correctly', () => {
    global.createRef = jest.fn().mockImplementation(() => ({
    current : {
      getContext : jest.fn()
    }
  }));
    expect(wrapper).toMatchSnapshot();
  });

});