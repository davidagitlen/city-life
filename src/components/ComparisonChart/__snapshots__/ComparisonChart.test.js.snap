// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonChart should match the snapshot with data passed in correctly 1`] = `undefined`;
